test/unit/captionsrenderer-test.js

```diff
@@ -21,9 +21,45 @@ define([
             new VTTCue(12, 15, 'EG: Hey, Jo...'),
             new VTTCue(13, 14, 'JB: Yeah?'),
             new VTTCue(13, 14, 'JP: Yeah?'),
-            new VTTCue(16, null, 'The End')
+            new VTTCue(16, 18, 'The End')
         ];
-        var currentNumCues = [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 2, 1, 2, 4, 4, 1, 1];
+        var currentNumCues = [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 2, 1, 2, 4, 4, 1, 1, 1, 1, 0, 0, 0];
+
+        for (var i = 0; i < currentNumCues.length; i += 1) {
+            assert.equal(captionsRenderer.getCurrentCues(allCues, i).length, currentNumCues[i], 'Invalid number of cues at position ' + i);
+        }
+    });
+
+    test('should show the correct number of cues at any given position in time with some missing end times', function (assert) {
+        var allCues = [
+            new VTTCue(0, undefined, 'HG: Morning, Rob.'),
```

Comment:
@hghazzi using `undefined` throws a TypeError in Chrome and sets `endTime` to `NaN` in Safari. A more realistic test is just using 0 (or `null`, which also sets `endTime` to 0 in Chrome and Safari), as that was the case in the stream where the bug was first discovered. Also, our parser returns 0 for invalid input.